packages/discovery-provider/src/queries/comments/utils.py

```diff
@@ -608,3 +671,105 @@ def fetch_related_entities(session, formatted_comments, current_user_id):
     tracks = get_tracks(session, track_ids, current_user_id)
 
     return users, tracks
+
+
+def get_comment_replies(
+    session,
+    parent_comment_ids,
+    current_user_id,
+    artist_id=None,
+    offset=None,
+    limit=None,
+):
+    """
+    Fetch replies for one or more parent comments.
+
+    Args:
+        session: Database session
+        parent_comment_ids: Single parent comment ID or list of parent comment IDs
+        current_user_id: ID of the user making the request
+        artist_id: ID of the track owner (optional)
+        offset: Pagination offset (optional)
+        limit: Pagination limit (optional)
+
+    Returns:
+        If parent_comment_ids is a list, returns a dict mapping parent_comment_id to list of formatted replies
+        If parent_comment_ids is a single ID, returns a list of formatted replies
+    """
+    from src.utils.helpers import encode_int_id
+
+    # Handle single parent_comment_id case
+    single_parent = not isinstance(parent_comment_ids, list)
+    if single_parent:
+        parent_comment_ids = [parent_comment_ids]
+
+    # Get base query components
+    base_query = get_base_comments_query(session, current_user_id)
+
+    # Build the query using batch_replies query type
+    replies_query = build_comments_query(
+        session=session,
+        query_type="batch_replies",
+        base_query=base_query,
+        current_user_id=current_user_id,
+        artist_id=artist_id,
+        parent_comment_ids=parent_comment_ids,
+    )
+
+    # Apply pagination if provided
+    if offset is not None:
+        replies_query = replies_query.offset(offset)
+    if limit is not None:
+        replies_query = replies_query.limit(limit)
+
+    # Execute the query
+    replies_results = replies_query.all()
+
+    # Process the results and organize replies by parent_comment_id
+    replies_map = {}
+    for reply_data in replies_results:
+        reply, react_count, mentions, parent_id = reply_data
+
+        # Format the reply
+        formatted_reply = {
+            "id": encode_int_id(reply.comment_id),
+            "entity_id": encode_int_id(reply.entity_id),
+            "entity_type": reply.entity_type,
+            "user_id": encode_int_id(reply.user_id) if not reply.is_delete else None,
+            "mentions": list(
+                map(
+                    lambda m: {k: v for k, v in m.items() if k != "is_delete"},
+                    [
+                        m
+                        for m in mentions
+                        if m["user_id"] is not None and m["is_delete"] is not True
+                    ],
+                )
+            ),
+            "message": reply.text if not reply.is_delete else "[Removed]",
+            "is_edited": reply.is_edited,
+            "track_timestamp_s": reply.track_timestamp_s,
+            "react_count": react_count,
+            "is_current_user_reacted": get_is_reacted(
```

Comment:
Is `get_is_reacted` also another one of those things that's going to run a separate query per item we process? :-D

Reply:
great find